impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -35,6 +35,415 @@ define(["./_focus"], function(focusUtil) {
     delete: 46
   };
 
+  function makeAccessibleSelect(elem) {
+    var listboxNode = elem;
+    listboxNode.setAttribute('role', 'listbox');
+    listboxNode.setAttribute('tabindex', '0');
+    listboxNode.setAttribute('aria-activedescendant', '');
+    listboxNode.setAttribute('aria-multiselectable', 'true');
+    var activeDescendant = listboxNode.getAttribute(
+        'aria-activedescendant'
+    );
+    var multiselectable = listboxNode.hasAttribute('aria-multiselectable');
+    var startRangeIndex = 0;
+    var keysSoFar = '';
+    var handleFocusChange = function () {};
+    var handleItemChange = function () {};
+    registerEvents();
+
+    /**
+     * @description
+     *  If there is no activeDescendant, focus on the first option
+     */
+    function setupFocus() {
+      activeDescendant = listboxNode.getAttribute(
+          'aria-activedescendant'
+      );
+      if (activeDescendant) {
+        var currentItem = document.getElementById(activeDescendant);
+        currentItem.classList.add('keyboard-navigation-focus');
+      }
+    }
+
+    /**
+     * @description
+     *  If there is no activeDescendant, focus on the first option
+     */
+    function setupBlur() {
+      activeDescendant = listboxNode.getAttribute(
+          'aria-activedescendant'
+      );
+      if (activeDescendant) {
+        var currentItem = document.getElementById(activeDescendant);
+        currentItem.classList.remove('keyboard-navigation-focus');
+      }
+    }
+
+    /**
+     * @description
+     *  Register events for the listbox interactions
+     */
+    function registerEvents() {
+      listboxNode.addEventListener('focus', setupFocus);
+      listboxNode.addEventListener('blur', setupBlur);
+      listboxNode.addEventListener('keydown', checkKeyPress);
+      listboxNode.addEventListener('click', checkClickItem);
+
+      if (multiselectable) {
+        listboxNode.addEventListener('mousedown', checkMouseDown);
+      }
+    }
+
+    /**
+     * Prevent text selection on shift + click for multi-select listboxes
```

Comment:
What do you mean by "text selection"?
And, isn't shift+click allowed in multiple-selection?